.circleci/config.yml

```diff
@@ -7,7 +7,9 @@ version: 2
 run_install_desired_npm: &run_install_desired_npm
   run:
     name: Install npm@6
-    command: sudo npm install -g npm@6
+    # we use yarn, since Node 6 does not support the latest npm by default, so
+    # use 5
+    command: sudo yarn global add npm@5
```

Comment:
@evans Node.js 6 will work fine with the latest `npm` (6.x), however it's the installation of the newer `npm` with the _older_ `npm@3` (which ships with Node.js 6) that is problematic (due to a bug in `npm@3`).

As a workaround, `yarn` should used to do the upgrade, which is why it's important for this to be re-introduced.  This was explained in the comment which was removed in 301e144d233c033b8e720fa4d5dafd, and it's important to leave that clarification there in the future so I've re-introduced it in 559b21d83a40aa349e1deb7f6d39d49b6f31e043, which also fixes the `Install npm@6` residue above (which is incorrect now, since `npm@5` is being used again)

Reply:
I should emphasize though: it should certainly be possible to switch to `npm@6` here, but `yarn` should still be used to install it.